src/server.c

```diff
@@ -2831,6 +2831,15 @@ void populateCommandStructure(struct redisCommand *c) {
             /* Translate the command string flags description into an actual
              * set of flags. */
             setImplicitACLCategories(sub);
+
+            /* If the command marks with CMD_SENTINEL, it exists in sentinel. */
+            if (!(sub->flags & CMD_SENTINEL) && server.sentinel_mode)
+                continue;
+
+            /* If the command marks with CMD_ONLY_SENTINEL, it only exists in sentinel. */
+            if (sub->flags & CMD_ONLY_SENTINEL && !server.sentinel_mode)
+                continue;
```

Comment:
can we move these to the populateCommandStructure entrance, instead of repeating them in both populateCommandTable and the subcommand loop in populateCommandStructure?